src/Raven.Client/Documents/Session/DocumentSession.Patch.cs

```diff
@@ -134,13 +151,53 @@ public override void ConvertToJavascript(JavascriptConversionContext context)
                     JsCompilationFlags.BodyOnly | JsCompilationFlags.ScopeParameter,
                     new LinqMethods(), extension));
 
-            var script = $"this.{pathScript}{adderScript}";
+            var parameters = new Dictionary<string, object>(extension.Parameters);
+            foreach (var kvp in parameters)
+            {
+                var newArg = $"{kvp.Key}_{_customCount}";
+                adderScript = adderScript.Replace(kvp.Key, newArg);
+                extension.Parameters.Remove(kvp.Key);
+                extension.Parameters[newArg] = kvp.Value;
+            }
+            _customCount++;
 
-            Advanced.Defer(new PatchCommandData(id, null, new PatchRequest
+            var patchRequest = new PatchRequest
             {
-                Script = script,
+                Script = $"this.{pathScript}{adderScript}",
                 Values = extension.Parameters
+            };
+
+            if (TryMergePatches(id, patchRequest) == false)
+            {
+                Advanced.Defer(new PatchCommandData(id, null, patchRequest, null));
+            }
+        }
+
+        private bool TryMergePatches(string id, PatchRequest patchRequest)
+        {
+            var patches = _deferredCommands.OfType<PatchCommandData>().ToList();
```

Comment:
Just do a `FirstOrDefault` search here, instead of creating the list and then searching it